src/components/Countdown/index.jsx

```diff
@@ -0,0 +1,64 @@
+import "./style.css";
+
+const hackathonStart = new Date("May 21, 2022 9:00:00").getTime();
+
+// update countdown every second
+var x = setInterval(function() {
+
+  var now = new Date().getTime();
+
+  // find the time between now and the count down date
+  var timeLeft = hackathonStart - now;
+
+  // calculations for days, hours, minutes, seconds
+  var days = Math.floor(timeLeft / (1000 * 60 * 60 * 24));
+  var hours = Math.floor((timeLeft % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60));
+  var minutes = Math.floor((timeLeft % (1000 * 60 * 60)) / (1000 * 60));
+  var seconds = Math.floor((timeLeft % (1000 * 60)) / 1000);
+
+   // if count down is finished, just display 0s
+  if (timeLeft < 0) {
+    clearInterval(x);
+    days = 0;
+    hours= 0;
+    minutes = 0;
+    seconds = 0;
+  }
+
+  // display time left 
+  document.getElementById("days-num").innerHTML = days;
+  document.getElementById("hours-num").innerHTML = hours;
+  document.getElementById("minutes-num").innerHTML = minutes;
+  document.getElementById("seconds-num").innerHTML = seconds;
+
+}, 1000);
+
+
+const Countdown = () => {
+  return <div id="countdown-wrapper">
```

Comment:
we want to initialize the setInterval after the component has mounted (rendered on the page) since its modifying page content so we can use another hook called useEffect for this!

this hook takes in a function and dependency array and then runs the function when the component renders and then every time a dependency changes. If we give it a function with a return value (optional), it will use this return value as a cleanup function (runs when the component is no longer being rendered)

```
useEffect(function() {
    do something
    return function() {
        do something when the component is deleted
    }
}, dependency array)
```
In our case, we only want to run a function once to initialize the countdown function so we can leave the dependency array blank. however, when the component is no longer rendering, we want to remove the interval so it's not left over so we should clear it in our clean up function. this might look something like
```
useEffect(function() {
    const timer = setInterval(...)
    return function() {
        clearInterval(timer)
    }
}, [])
```